src/components/Footer/Footer.tsx

```diff
@@ -0,0 +1,83 @@
+import React, { useContext, useEffect } from 'react';
+import cn from 'classnames';
+import { TodosContext } from '../Store/Store';
+import { FilterParams } from '../../types/FilterParams';
+
+export const Footer: React.FC = React.memo(() => {
+  const {
+    todos,
+    filter,
+    setFilter,
+    deleteTodo,
+    pressClearAll,
+    setPressClearAll,
+  } = useContext(TodosContext);
+
+  const itemsLeft = todos.filter(todo => !todo.completed).length;
+  const hasCompleted = todos.some(todo => todo.completed);
+
+  const handleClearCompleted = () => {
+    setPressClearAll(true);
+  };
+
+  useEffect(() => {
+    if (pressClearAll) {
+      const completedTodos = todos.filter(todo => todo.completed);
+
+      Promise.all(completedTodos.map(todo => deleteTodo(todo.id)));
+    }
+  // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [pressClearAll, todos]);
```

Comment:
this logic shouldn't be in `useEffect` I suppose.
and do not ignore eslint warning, it is very important